### ISSUE TITLE
Open questions regarding NamespaceMap and their relation to serialization

### ISSUE THREAD

**Issue by maxhbr** [spdx/spdx-3-model#483]
Some examples where `NamespaceMap` has issues. Lets assume that the collections have the following Namespace Maps
* Collection1
  * "key1": "some.url.invalid/123"
  * "key2": "another.url.invalid"
* Collection2
  * "key1": "a_tihrd.url.invalid/123abc"
  * "key3": "the.fourth.url.test"

## P1 cascaded Collections
`Collection1` which contains `Collection2` and `Collection2` contains `Element`, or as a graph:
```mermaid
graph TD;
    Collection1-->Collection2;
    Collection2-->Element;
```

- What NamespaceMap applies to `Collection2`?
- What NamespaceMap applies to `Element`?
- What is the single namespace Map which should be used in a serialization that only support one consistent serialization?

## P2 multiple top level Collections
```mermaid
graph TD;
    Collection1-->Element;
    Collection2-->Element;
```

- What NamespaceMap applies to `Element`?
- What is the single namespace Map which should be used in a serialization that only support one consistent serialization?

## P3 collections that include each other
```mermaid
graph TD;
    Collection1-->Collection2;
    Collection2-->Collection1;
```

- What NamespaceMap applies to `Collection1`?
- What NamespaceMap applies to `Collection2`?
- What is the single namespace Map which should be used in a serialization that only support one consistent serialization?

### P3b collections that contain themself
```mermaid
graph TD;
    Collection1-->Collection1;
```

**Comment by davaya** [comment]
+1

Note that these examples all relate to defining NamespaceMap in the model, and when Element has an SpdxId multiple keys could apply to it, but only if the URLs overlap, which they don't in the example.  If Element starts with "some", only key1 can apply to it regardless of what collections it's in.

When NamespaceMap is defined in serialization there is no problem with either the single Element shown or multiple Elements.

Assume Collection1 has three elements some.url.invalid/123/E1, some.url.invalid/123/E2 and another.url.invalid/E3.
Assume Collection 2 has three elements a_tihrd.url.invalid/123abc/E4, the.fourth.url.test/E5, and the.fourth.url.test/E6.

With this small collection of elements there is no point in serializing NamespaceMap keys for URLs that appear only once. But E1/E2 and E5/E6 can be shortened by defining "key1" and "key4" in the payload, or "m1" and "m2" for those URIs because the specific keys don't exist anywhere in the model, just the Element IDs.

In a JSON document the namespace map would be {"m1": "some.url.invalid/123/", "m2": "the.fourth.url.test/"}, and in a LawyerScript document it could be:

**NamespaceMap:**
  **m1:** some.url.invalid/123/
  **m2:** the.fourth.url.test/

**Comment by goneall** [comment]
@maxhbr - pinging you again on this - let me know if this can be deferred until after RC2 or if it is something we should discuss

**Comment by goneall** [comment]
Pushing to post RC2 unless @maxhbr objects

**Comment by goneall** [comment]
@maxhbr  - Since I haven't heard from you on this after 2 pings, I'm going to close this issue.  If you think it is still an issue, please re-open
